include/fty_common_mlm_client.h

```diff
@@ -0,0 +1,126 @@
+         * This method handles the infinite loop of reception messages from the
+         * agent's pipe and malamute connection and forwards them to the appropriate
+         * handle*() callbacks.
+         */
+        void mainloop();
+
+    protected:
+        /**
+         * \brief Constructor.
+         *
+         * This will create and connect the mlm_client_t object. The agent must
+         * handle the configuration of the consumption and production parts of the
+         * connection.
+         *
+         * \param pipe Pipe of the zactor_t
+         * \param endpoint Endpoint to connect to
+         * \param address Name of the established connection
+         * \param pollerTimeout Timeout of poller and rough interval betwenn tick() invocations (-1 to disable)
+         * \param connectionTimeout Timeout for connection attempt
+         */
+        MlmClient(zsock_t *pipe, const char *endpoint = NULL, const char *address = NULL, int pollerTimeout = -1, int connectionTimeout = 5000);
+        /**
+         * \brief connect to broker malamute
+         * \return false to stop the agent, true otherwise.
+         * \param endpoint Endpoint to connect to
+         * \param address Name of the established connection
+         * \param connectionTimeout Timeout for connection attempt
+         */
+        virtual void connect(const char *endpoint, const char *address, int connectionTimeout = 5000);
+        /**
+         * \brief Periodic callback, if enabled.
+         * \return false to stop the agent, true otherwise.
+         */
+        virtual bool tick() { return true; }
+
+        /**
+         * \brief Callback for stream messages. The callback DOESN'T take ownership of the message.
+         * \return false to stop the agent, true otherwise.
+         */
+        virtual bool handleStream(zmsg_t *message) { return true; }
+
+        /**
+         * \brief Callback for pipe messages. The callback DOESN'T take ownership of the message.
+         * \return false to stop the agent, true otherwise.
+         */
+        virtual bool handlePipe(zmsg_t *message);
+
+        /**
+         * \brief Callback for mailbox messages. The callback DOESN'T take ownership of the message.
+         * \return false to stop the agent, true otherwise.
+         */
+        virtual bool handleMailbox(zmsg_t *message) { return true; }
+
+        /**
+         * \brief Callback for "other" messages. The callback DOESN'T take ownership of the message.
+         * \return false to stop the agent, true otherwise.
+         */
+        virtual bool handleOther(zmsg_t *message, void *which) { return true; }
+
+        /**
+         * \brief Create/get the zpoller_t* object for the agent mainloop.
+         * \return The mainloop poller object.
+         */
+        virtual zpoller_t* zpoller();
+
+        mlm_client_t *m_client;
+        zsock_t *m_pipe;
+        int64_t m_lastTick;
+        int m_pollerTimeout;
+        int m_connectionTimeout;
```

Comment:
Consider removing `m_connectionTimeout` as its value isn't required once connection is complete.